worker-core/src/main/java/com/hpe/caf/worker/core/WorkerExecutor.java

```diff
@@ -78,23 +78,59 @@ public void executeTask(final TaskMessage tm, final TaskInformation taskInformat
     }
 
     /**
-     * Decide whether the message is to be forwarded or discarded.
+     * Decide whether the message is to be forwarded, executed, or discarded.
      *
      * @param tm the task message
      * @param taskInformation the reference to the message this task arrived on
+     * @param poison flag indicating if the message is a poison message
      * @param headers the map of key/value paired headers to be stamped on the message
+     * @param codec the Codec that can be used to serialise/deserialise data
+     * @param jobStatus the job status as returned by the status check URL
      */
-    public void forwardTask(final TaskMessage tm, final TaskInformation taskInformation, Map<String, Object> headers) throws TaskRejectedException
+    public void forwardTask(final TaskMessage tm, final TaskInformation taskInformation, final boolean poison,
```

Comment:
Given the change in functionality we should perhaps think about renaming this method - perhaps `handleDivertedTask`?